simulator/func_sim/alu.h

```diff
@@ -285,6 +286,14 @@ struct ALU
         instr->v_dst[0] = ( (instr->v_src[0] >> pack_width) | (instr->v_src[1] & (bitmask<T>(pack_width) << pack_width)));
     }
 
+    template<typename T> static
+    void bclri( Instr* instr )
+    {
+        uint64_t index = shamt_imm( instr) & (XLEN - 1);
+        uint64_t mask = ~(static_cast<uint64_t>( 1) << index);
```

Comment:
That won't work for RV128, it will cause overflow. Could you please add tests and fix it? I think you need to use `RegisterUInt` here instead of `uint64_t`?

Reply:
It looks like there is extra bit left for RV128 shamt. I will add RV128 support, but there are only RV32 and RV64 encodings for this instruction, according to specification.

Reply:
![image](https://user-images.githubusercontent.com/70326825/142249074-ad7e6698-cdf8-4431-b880-e5b27bf13fda.png)